EventListener/UserContextSubscriber.php

```diff
@@ -146,18 +153,24 @@ public function onKernelResponse(FilterResponseEvent $event)
                 return;
```

Comment:
@supasteev0 we added this sanity check about the hash header in a different branch, while you did your pull request. i decided even if we don't auto-set the vary header, we still do the sanity check (the typical situation would be that a session timed but the hash is still cached)

do you agree it makes sense to then still not cache the response? i check the new flag you add only after this sanity check.